File: frontend/src/types/ticket.ts
export type TicketCategory =
  | 'Hardware'
  | 'Software'
  | 'Network'
  | 'Access Request'
  | 'Asset Request'
  | 'Security'
  | 'Other'

export type TicketPriority = 'Low' | 'Medium' | 'High' | 'Critical'

export type TicketStatus =
  | 'Open'
  | 'In Progress'
  | 'Waiting for User'
  | 'Resolved'
  | 'Closed'

export interface Ticket {
  id: number
  ticket_number: string
  title: string
  description: string
  category: TicketCategory
  priority: TicketPriority
  status: TicketStatus
  created_by: number
  assigned_to: number | null
  created_at: string
  updated_at: string
  resolution_notes?: string | null
  legacy_event_id?: number | null
}

export interface TicketComment {
  id: number
  ticket_id: number
  user_id: number
  body: string
  created_at: string
}

export interface TicketDetail extends Ticket {
  comments: TicketComment[]
}

export interface TicketCreate {
  title: string
  description: string
  category: TicketCategory
  priority: TicketPriority
}

export interface TicketUpdate {
  title: string
  description: string
  category: TicketCategory
  priority: TicketPriority
  resolution_notes?: string | null
}

File: app/constants/ticket.py
"""
Ticket enums for categories, priorities, and workflow status.
"""

from enum import Enum


class TicketCategory(str, Enum):
    hardware = "Hardware"
    software = "Software"
    network = "Network"
    access_request = "Access Request"
    asset_request = "Asset Request"
    security = "Security"
    other = "Other"


class TicketPriority(str, Enum):
    low = "Low"
    medium = "Medium"
    high = "High"
    critical = "Critical"


class TicketStatus(str, Enum):
    open = "Open"
    in_progress = "In Progress"
    waiting_for_user = "Waiting for User"
    resolved = "Resolved"
    closed = "Closed"


# Terminal statuses — used for viewer "resolved" views and metrics.
TERMINAL_STATUSES = frozenset({TicketStatus.resolved.value, TicketStatus.closed.value})
